### PR TITLE
Add support for firefox mac installer. Update web_ui pubspec for http.wq

### PR DESCRIPTION
**Pull request by ferhatb**

### CODE REVIEW

File: lib/web_ui/dev/common.dart [hunk 1]
```diff
@@ -150,12 +161,16 @@ class _MacBinding implements PlatformBinding {
...
-      'https://download-installer.cdn.mozilla.net/pub/firefox/releases/${version}/mac/en-US/firefox-${version}.dmg';
+    'https://download-installer.cdn.mozilla.net/pub/firefox/releases/${version}/mac/en-US/'
+        '${getFirefoxDownloadFilename(version)}';
 
   @override
-  String getFirefoxExecutablePath(io.Directory versionDir) {
-    throw UnimplementedError();
-  }
+  String getFirefoxDownloadFilename(String version) =>
+      'Firefox ${version}.dmg';
```
Comment (nturgut) [inline lib/web_ui/dev/common.dart]: question: is there a space between name and version?

File: lib/web_ui/dev/firefox_installer.dart [hunk 2]
```diff
@@ -188,16 +195,20 @@ class FirefoxInstaller {
     ));
 
     final io.File downloadedFile =
-        io.File(path.join(versionDir.path, 'firefox-${version}.tar.bz2'));
-    await download.stream.pipe(downloadedFile.openWrite());
+        io.File(path.join(versionDir.path, PlatformBinding.instance.getFirefoxDownloadFilename(version)));
+    io.IOSink sink = downloadedFile.openWrite();
```
Comment (nturgut) [inline lib/web_ui/dev/firefox_installer.dart]: Thanks! So if we don't do this, we are leaking.

I think we made this error in a few more places :) https://github.com/flutter/engine/blob/master/lib/web_ui/dev/chrome_installer.dart#L190

I'll fix them!

File: lib/web_ui/dev/firefox_installer.dart [hunk 3]
```diff
@@ -188,16 +195,20 @@ class FirefoxInstaller {
...
     return downloadedFile;
   }
 
-  /// Uncompress the downloaded browser files.
+  /// Uncompress the downloaded browser files for operating systems that
+  /// use a zip archive.
   /// See [version].
   Future<void> _uncompress(io.File downloadedFile) async {
-    final io.ProcessResult unzipResult = await io.Process.run('tar', <String>[
+    io.ProcessResult unzipResult = await io.Process.run('tar', <String>[
```
Comment (nturgut) [inline lib/web_ui/dev/firefox_installer.dart]: nit: why not final?

File: lib/web_ui/dev/common.dart (hunk 1 above)
Comment (ferhatb) [reply lib/web_ui/dev/common.dart]: Yes

File: lib/web_ui/dev/firefox_installer.dart (hunk 2 above)
Comment (ferhatb) [reply lib/web_ui/dev/firefox_installer.dart]: Thank you!

File: lib/web_ui/dev/firefox_installer.dart (hunk 3 above)
Comment (ferhatb) [reply lib/web_ui/dev/firefox_installer.dart]: Done.

File: lib/web_ui/dev/firefox_installer.dart [hunk 4]
```diff
@@ -67,8 +67,9 @@ Future<BrowserInstallation> getOrInstallFirefox(
 }) async {
   // These tests are aimed to run only on the Linux containers in Cirrus.
   // Therefore Firefox installation is implemented only for Linux now.
-  if (!io.Platform.isLinux) {
-    throw UnimplementedError();
+  if (!io.Platform.isLinux && !io.Platform.isMacOS) {
```
Comment (nturgut) [inline lib/web_ui/dev/firefox_installer.dart]: thank you!

**Comment by ferhatb** [comment]
Merging on infra failure.

**Pull request merged**